src/test/java/com/pusher/rest/PusherTest.java

```diff
@@ -96,6 +98,24 @@ protected String serialise(Object data) {
     }
 
     @Test
+    public void batchEvents() throws IOException {
+        context.checking(new Expectations() {{
+            oneOf(httpClient).execute(
+                with(
+                  field(
+                    "batch",
+                    "[{\"channel\": \"my-channel\", \"name\": \"event-name\", \"data\": \"data\"},{\"channel\": \"my-channel\", \"name\": \"event-name\", \"data\": \"data2\", \"socket_id\": \"22.33\"}]"
```

Comment:
The batch contents aren't double encoded, so you should be matching against a `List` of two `Map`s here, not a string.